docstore/driver/codec.go

```diff
@@ -144,6 +144,14 @@ func encode(v reflect.Value, enc Encoder) error {
 		}
 		return nil
 	}
+	if reflect.PtrTo(v.Type()).Implements(protoMessageType) {
```

Comment:
Using a struct with a protocol buffer field (as opposed to a pointer to a protocol buffer field) failed before this change; see #2791.

Reply:
@jba this change looks OK?

Reply:
Yes.